File: src/#1110/solution.go
package solution

type TreeNode struct {
	Val         int
	Left, Right *TreeNode
}

func delNodes(root *TreeNode, to_delete []int) []*TreeNode {
	res := []*TreeNode{}
	newRoot:=helper(root,to_delete,&res)
	if newRoot!=nil{
		res=append(res,newRoot)
	}
	return res

}
func helper(root *TreeNode, toDel []int, tn *[]*TreeNode) *TreeNode {
	if root == nil {
		return nil
	}
	root.Left = helper(root.Left, toDel, tn)
	root.Right = helper(root.Right, toDel, tn)

	for _, v := range toDel {
		if v == root.Val {
			if root.Left != nil {
				*tn = append(*tn, root.Left)
			}
			if root.Right != nil {
				*tn = append(*tn, root.Right)
			}
			return nil
		}
	}
	return root
}

File: src/#1110/solution.js
function TreeNode(val, left, right) {
  this.val = val === undefined ? 0 : val
  this.left = left === undefined ? null : left
  this.right = right === undefined ? null : right
}

var delNodes = function (root, to_delete) {
  let res = []

  let newRoot = helper(root, to_delete)
  if (newRoot) {
    res.push(newRoot)
  }
  return res

  function helper(root, toDel) {
    if (!root) return null

    root.left = helper(root.left, toDel)
    root.right = helper(root.right, toDel)
    for (let v of toDel) {
      if (v === root.val) {
        if (root.left) {
          res.push(root.left)
        }
        if (root.right) {
          res.push(root.right)
        }
        return null
      }
    }
    return root
  }
}
